specification/cognitiveservices/data-plane/LUIS/Authoring/stable/v2.0/LUIS-Authoring.json

```diff
@@ -9818,6 +10149,40 @@
           "type": "string"
         }
       }
+    },
+    "AzureAccountInfoList": {
+      "description": "List of Azure account info objects.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/AzureAccountInfoObject"
+      }
+    },
+    "AzureAccountInfoObject": {
+      "description": "Defines the azure account information object.",
+      "type": "object",
+      "required": [
+        "azureSubscriptionId",
+        "resourceGroup",
+        "accountName"
+      ],
+      "properties": {
+        "azureSubscriptionId": {
+          "description": "The id for the azure subscription.",
+          "type": "string"
+        },
+        "resourceGroup": {
+          "description": "The azure resource group name.",
+          "type": "string"
+        },
+        "accountName": {
+          "description": "The azure account name.",
+          "type": "string"
+        }
+      }
+    },
+    "PackageBinaryStreamObject": {
```

Comment:
@yangyuan can you advise on what needs to be done here? The API returns a byte array (binary octet stream) in the response. How could we document its response here?

For context, we do not want to give the users visibility on how to deserialize the object. We just instruct our users to call that API to export the model and import it back in a different place. So we want to expose the minimum amount of details about the response format.

We tried have this as type array, type string & type object but the three fail giving the following error:

FATAL: System.Collections.Generic.KeyNotFoundException: Please specify a field with type of System.Byte[] to deserialize the file contents to
   at AutoRest.Modeler.OperationBuilder.VerifyFirstPropertyIsByteArray(CompositeType serviceType) in C:\Users\ci\AppData\Local\Temp\PUBLISHu1dw1\44_20171214T003644\autorest.modeler\src\OperationBuilder.cs:line 441
   at AutoRest.Modeler.OperationBuilder.TryBuildStreamResponse(HttpStatusCode responseStatusCode, OperationResponse response, Method method, List`1 types, IModelType headerType) in C:\Users\ci\AppData\Local\Temp\PUBLISHu1dw1\44_20171214T003644\autorest.modeler\src\OperationBuilder.cs:line 422
   at AutoRest.Modeler.OperationBuilder.BuildResponses(Method method, CompositeType headerType) in C:\Users\ci\AppData\Local\Temp\PUBLISHu1dw1\44_20171214T003644\autorest.modeler\src\OperationBuilder.cs:line 336
   at AutoRest.Modeler.OperationBuilder.BuildMethod(HttpMethod httpMethod, String url, String methodName, String methodGroup) in C:\Users\ci\AppData\Local\Temp\PUBLISHu1dw1\44_20171214T003644\autorest.modeler\src\OperationBuilder.cs:line 195
   at AutoRest.Modeler.SwaggerModeler.BuildMethod(HttpMethod httpMethod, String url, String name, Operation operation) in C:\Users\ci\AppData\Local\Temp\PUBLISHu1dw1\44_20171214T003644\autorest.modeler\src\SwaggerModeler.cs:line 366
   at AutoRest.Modeler.SwaggerModeler.Build(ServiceDefinition serviceDefinition) in C:\Users\ci\AppData\Local\Temp\PUBLISHu1dw1\44_20171214T003644\autorest.modeler\src\SwaggerModeler.cs:line 105
   at AutoRest.Modeler.Program.<ProcessInternal>d__2.MoveNext() in C:\Users\ci\AppData\Local\Temp\PUBLISHu1dw1\44_20171214T003644\autorest.modeler\src\Program.cs:line 60
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NewPlugin.<Process>d__15.MoveNext()
FATAL: nodejs/imodeler1 - FAILED
FATAL: Error: Plugin imodeler1 reported failure.
Process() cancelled due to exception : Plugin imodeler1 reported failure.
  Error: Plugin imodeler1 reported failure.

Reply:
Didn't test but I think it should be `"format": "binary"`:
```json
"PackageBinaryStreamObject": {
  "description": "The binary stream of the application package.",
  "type": "string",
  "format": "binary"
}
```
You should also add `"content":  "application/octet-stream"` in response.

About "we do not want to give the users visibility on how to deserialize the object."
Why not just exclude this from swagger?

Reply:
@yangyuan format binary is also not working, giving same error.

On adding content type, I already have:
```
"produces": [
          "application/octet-stream"
        ]
```
This is what we use in all our APIs to specify content type, is this what you mean? If there is another place to add `"content": "application/octet-stream"`, please advise where.

Reply:
And to answer your suggestion about excluding from swagger, we want the API to be visible for our users to call it but we do not want them to parse the response.  Just call this API to export the app and import it elsewhere.

Reply:
Weird, `format binary` is pretty common, surprisingly it's not supported.

But if that's the case, please refer to https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/ComputerVision/stable/v1.0/ComputerVision.json
and use `"type": "file"`

For `octet-stream`, `produces` should be good enough.

Reply:
@yangyuan still failing with same error. As per the ComputerVision example, I added `"type": "file"` in the response schema and added `"body": "{binary}"` to the response examples parameters and body, still no luck.

Reply:
@yangyuan a gentle reminder.

Reply:
Please give me the exact command which causes the given error.
I started to think this might be other issues.

I checked travis log, didn't see this error.

Reply:
I am not running any manual commands, I see the error in the automated build checks:
https://github.com/Azure/azure-rest-api-specs/pull/4524/checks?check_run_id=37479764

Reply:
I pulled your branch, and run
`autorest specification/cognitiveservices/data-plane/LUIS/Authoring/ --go`
And I can see the exact same error.

Then I modify the `"responses"` of `"/package/{appId}/versions/{versionId}/gzip"` `"/package/{appId}/slot/{slotName}/gzip"` into below

```
        "responses": {
          "200": {
            "description": "The GZip binary stream of the published application package.",
            "schema": {
              "type": "file"
            }
          }
        }
```

And the problem gone.
Which means `"type": "file"` is valid, and I suspect `"format": "binary"` can work too.

Reply:
I debugged a little bit, this is the solution.

```
        "produces": [
          "application/octet-stream",
          "application/json"
        ],
```

The problem is you define the response can only be `application/octet-stream`, so autorest try to assert that all response should be binary format and gives error. :)